src/overlays/actors/ovl_En_Takara_Man/z_en_takara_man.c

```diff
@@ -27,25 +36,199 @@ const ActorInit En_Takara_Man_InitVars = {
+                 globalCtx->actorCtx.flags.chest); // "Bun! %x" (needs a better translation)
+    globalCtx->actorCtx.flags.chest = 0;
+    gSaveContext.dungeonKeys[gSaveContext.mapIndex] = -1;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06004FE0, &D_06000498, this->limbDrawTbl, this->transitionDrawTbl,
+                     10);
+    thisx->posRot2.pos = thisx->posRot.pos;
+    this->pos = thisx->posRot.pos;
+    thisx->posRot.pos.x = 133.0f;
+    thisx->posRot.pos.y = -12.0f;
+    thisx->posRot.pos.z = 102.0f;
+    Actor_SetScale(&this->actor, 0.013f);
+    this->eyePos = 90.0f;
+    this->curRoomNum = thisx->room;
+    thisx->room = 0xFF;
+    thisx->posRot.rot.y = thisx->shape.rot.y = -0x4E20;
+    thisx->unk_1F = 1;
+    this->actionFunc = EnTakaraMan_InitAnimation;
+}
+
+void EnTakaraMan_InitAnimation(EnTakaraMan* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000498.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000498, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f); // cast required
+    if (!this->unk_214) {
+        this->actor.textId = 0x6D; // "Open the chest and..Surprise! ... 10 Rupees to play .. Yes/No"
+        this->dialogState = 4;
+    }
+    this->actionFunc = func_80B1778C;
+}
+
+void func_80B1778C(EnTakaraMan* this, GlobalContext* globalCtx) {
+    s16 yawDiff;
+    s16 yawDiffTemp;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_8002F194(&this->actor, globalCtx) && this->dialogState != 6) {
+        if (!this->unk_214) {
+            this->actionFunc = EnTakaraMan_SetChoiceTextID;
+        } else {
+            this->actionFunc = func_80B17B14;
+        }
+    } else {
+        yawDiffTemp = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+        if (globalCtx->roomCtx.curRoom.num == 6 && !this->unk_21A) {
+            this->actor.textId = 0x6E; // "Great! You are a real gambler!"
+            this->unk_21A = 1;
+            this->dialogState = 6;
+        }
+
+        if (!this->unk_21A && this->unk_214) {
+            if (Flags_GetSwitch(globalCtx, 0x32)) {
+                this->actor.textId = 0x84; // "Thanks a lot!"
+                this->dialogState = 5;
+            } else {
+                this->actor.textId = 0x704C; // "With that key, proceed to the room ahead. Go, go!"
+                this->dialogState = 6;
+            }
+        }
+
+        yawDiff = ABS(yawDiffTemp);
+        if (yawDiff < 0x4300) {
+            if (globalCtx->roomCtx.curRoom.num != this->curRoomNum) {
+                this->actor.flags &= ~1;
+                this->unk_218 = 0;
+            } else {
+                if (!this->unk_218) {
+                    this->actor.flags |= 1;
+                    this->unk_218 = 1;
+                }
+                func_8002F2CC(&this->actor, globalCtx, 100.0f);
+            }
+        }
+    }
+}
+
+void EnTakaraMan_SetChoiceTextID(EnTakaraMan* this, GlobalContext* globalCtx) {
```

Comment:
Not a huge fan of this name either, tho it's already better. Maybe we can come up with something more explicit or leave it for more documentation later?